cirq-core/cirq/ops/kraus_channel.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, Dict, Iterable, Tuple, Union
+import numpy as np
+
+from cirq import linalg, protocols, value
+from cirq._compat import proper_repr
+from cirq.ops import raw_types
+
+
+class KrausChannel(raw_types.Gate):
+    """A generic channel that can record the index of its selected operator.
+
+    Args:
+        kraus_ops: a list of Kraus operators, formatted as numpy array.
```

Comment:
Kraus operators don't have to be square matrices. Non-square matrices appear when you have channels between Hilbert spaces of different dimension, e.g. a channel that consumes a qutrit and produces a qubit.

At minimum we should be explicit in comments and instantiation checks about whether or not non-square Kraus operators are supported. Better yet, let's make this work for non-square Kraus operators!

Reply:
Resolved. See answer to a different comment.

Reply:
Clarified this in the docstring, and added a TODO for adding qudit / non-square operator support.